meilisearch-types/src/document_formats.rs

```diff
@@ -130,30 +165,46 @@ pub fn read_ndjson(input: impl Read, writer: impl Write + Seek) -> Result<usize>
     Ok(count as usize)
 }
 
-/// Reads JSON from input and write an obkv batch to writer.
-pub fn read_json(input: impl Read, writer: impl Write + Seek) -> Result<usize> {
-    let mut builder = DocumentsBatchBuilder::new(writer);
-    let reader = BufReader::new(input);
-
-    #[derive(Deserialize, Debug)]
-    #[serde(transparent)]
-    struct ArrayOrSingleObject {
-        #[serde(with = "either::serde_untagged")]
-        inner: Either<Vec<Object>, Object>,
-    }
-
-    let content: ArrayOrSingleObject =
-        serde_json::from_reader(reader).map_err(Error::Json).map_err(|e| (PayloadType::Json, e))?;
+/**
+ * https://serde.rs/stream-array.html
+ * https://github.com/serde-rs/json/issues/160
+ */
+fn array_each<'de, D, T, F>(deserializer: D, f: F) -> std::result::Result<io::Result<u64>, D::Error>
```

Comment:
Could you add a simple description of what this function do?
Also, it looks like it will reject empty arrays of documents, I am not sure we want that, it is not an issue to send zero documents to Meilisearch.